index.js

```diff
@@ -2,14 +2,80 @@ const Vis = require("./graphvis");
 const { Module, render } = require("./visRenderer");
 
 let Sequelize;
-const relationship = ({ source, target, associationType, as }) => {
-  const typeString = {
-    BelongsTo: `[arrowtail=crow, arrowhead=none, dir=both, arrowsize=0.60]`,
-    BelongsToMany: `[arrowtail=crow, arrowhead=crow, dir=both, arrowsize=0.60]`
-  }[associationType];
-  if (typeString) {
-    return `"${source.name}" -> "${target.name}" ${typeString}`;
-  }
+
+const relationships = (associations, arrowShapes = {}, arrowSize = 0.6) => {
+  const mappings = [];
+
+  const mapper = {
+    associationTypes: new Map([
+      ['BelongsToMany', {
+        typeArrowShapes: arrowShapes['BelongsToMany']
+          ? arrowShapes['BelongsToMany']
+          : ['none', 'crow'],
+        items: []
+      }],
+      ['BelongsTo', {
+        typeArrowShapes: arrowShapes['BelongsTo']
+          ? arrowShapes['BelongsTo']
+          : ['crow', 'none'],
+        items: []
+      }],
+      ['HasMany', {
+        typeArrowShapes: arrowShapes['HasMany']
+          ? arrowShapes['HasMany']
+          : ['none', 'crow'],
+        items: []
+      }],
+      ['HasOne', {
+        typeArrowShapes: arrowShapes['HasOne']
+          ? arrowShapes['HasOne']
+          : ['none', 'none'],
+        items: []
+      }],
+    ]),
+  };
+
+  // Separation of association by type to process in the right order
+  associations.forEach(association =>
```

Comment:
What if we consider something like

```
const associationsByType = _.groupBy(associations, association => association.associationType)
```

Reply:
Is that lodash? That would probably work great, but it would add a dependency. I don't think there's a direct equivalent function in vanilla JS, is there?

Reply:
You're correct, I don't think there is. But I think it's very expressive, and I support adding the dependency

Reply:
Ok, what do you think of this option instead of installing lodash:

```
const groupedAssociations = associations.reduce((result, association) => {
  if (typeof result[association.associationType] === 'undefined')
    result[association.associationType].push(association);
  else
    result[association.associationType] = [association];
}, {});
```

It accomplishes the same thing in just a few extra rows without extra dependencies.

If you'd rather add lodash, I can do that, too, but I feel that fewer dependencies is typically better.

Reply:
I used the above solution (except I fixed the glaring error of not returning `result` after modifying it :sweat_smile:) and it's pretty good!